bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/StaticReferenceRewriteTransformerFactory.java

```diff
@@ -224,14 +230,24 @@ private String handleMatchingPatternAttribute(Pattern pattern, String attrValue)
                 if (url.startsWith(prefix)) {
                     // prepend host
                     url = prependHostName(url);
-                    m.appendReplacement(sb, Matcher.quoteReplacement(url));
-                    // First prefix match wins
+                    // Added check to determine whether the existing host has to be replaced
+                    if (BooleanUtils.isTrue(this.replaceHost)) {
```

Comment:
Same thing -- could just be `if (this.replaceHost)` if `replaceHost` is a primitive.